jumpscale/sals/nginx/nginx.py

```diff
@@ -318,16 +326,45 @@ def generate_certificates(self, retries=6):
         if self.domain:
             if self.key_path and self.cert_path and self.fullchain_path:
                 # only use install command if an existing key and certificate were set
-                cmd = self.certbot.install_cmd
+                self.install_certifcate(retries=retries)
             else:
-                cmd = self.certbot.run_cmd
-
-            for _ in range(retries):
-                rc, out, err = j.sals.process.execute(cmd)
-                if rc > 0:
-                    j.logger.error(f"Generating certificate failed {out}\n{err}")
-                else:
-                    break
+                self.obtain_and_install_certifcate(retries=retries)
+
+    def install_certifcate(self, retries=6):
+        """Construct and Execute install certificate command
+        Alternative to certbot install
+
+        Args:
+            retries (int, optional): Number of retries Certbot will try to install the certificate if failed. Defaults to 6.
+        """
+        cmd = self.certbot.install_cmd
+        j.logger.debug(f"Execute: {' '.join(cmd)}")
+        for _ in range(retries):
+            rc, out, err = j.sals.process.execute(cmd)
+            if rc > 0:
+                j.logger.error(f"Generating certificate failed {out}\n{err}")
+            else:
+                break
+
+    def obtain_and_install_certifcate(self, retries=6):
+        """Construct and Execute run certificate command,This will issue a new certificate managed by Certbot
+        Alternative to certbot run
+
+        Args:
+            retries (int, optional): Number of retries Certbot will try to install the certificate if failed. Defaults to 6.
+        """
+        # To make sure that the old info clear
+        self.certbot.key_path = ""
```

Comment:
I think no need to alter current object state by removing key and cert paths, Certbot run command will ignore these path arguments anyway.

```suggestion
```

Reply:
Ok, I will remove them in the next commit